google-cloud-graalvm-samples/graalvm-samples-client-library/bigquery-sample/src/main/java/com/example/BigQuerySampleApplication.java

```diff
@@ -0,0 +1,173 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryError;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Dataset;
+import com.google.cloud.bigquery.DatasetInfo;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.InsertAllRequest;
+import com.google.cloud.bigquery.InsertAllResponse;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.StandardTableDefinition;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableDefinition;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableInfo;
+import com.google.cloud.bigquery.TableResult;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Sample application demonstrating BigQuery operations.
+ *
+ * <p>Note: This application will create a BigQuery dataset in your GCP project.
+ * You can delete this by viewing BigQuery in Cloud Console
+ * https://console.cloud.google.com/bigquery or by uncommenting the call to `deleteDataset(..)`
+ * made in main().
+ */
+public class BigQuerySampleApplication {
+
+  private static final String DATASET_ID = "graalvm_test_dataset";
+
+  private static final String TABLE_ID = "graalvm_test_table";
+
+  private static final Schema TABLE_SCHEMA = Schema.of(
+      Field.of("id", StandardSQLTypeName.STRING),
+      Field.of("age", StandardSQLTypeName.INT64));
+
+  /**
+   * Entrypoint to the application.
+   */
+  public static void main(String[] args) throws InterruptedException {
+    BigQuery bigQuery = BigQueryOptions.getDefaultInstance().getService();
+
+    if (!hasDataset(bigQuery, DATASET_ID)) {
+      createDataset(bigQuery, DATASET_ID);
+    }
+
+    String tableName = TABLE_ID + "_"
+        + UUID.randomUUID().toString().replace("-", "");
+
+    createTable(bigQuery, DATASET_ID, tableName, TABLE_SCHEMA);
+    insertTestRecord(bigQuery, DATASET_ID, tableName);
+    queryTable(bigQuery, DATASET_ID, tableName);
+
+    // Clean up resources.
+    deleteTable(bigQuery, DATASET_ID, tableName);
+
+    // Uncomment this to delete the created dataset.
+    // deleteDataset(bigQuery, DATASET_ID);
```

Comment:
Just curious, why not delete by default?

Reply:
I wanted to do this, but unfortunately it makes the CI flaky since deleting datasets is a slow/eventually-consistent operation.

So I wanted to keep it commented to make the CI image tests more reliable.